### PR TITLE
Android Embedding PR 12: Add lifecycle methods to FlutterActivity.

### PR DESCRIPTION
**Pull request by matthew-carroll**
This work will be applied atop PR11, so don't worry about `FlutterEngine` not being instantiated in `FlutterFragment`:
https://github.com/flutter/engine/pull/7972

### CODE REVIEW

**Comment by matthew-carroll** [comment]
I did a force push after rebasing against master and resolving conflicts from PR 11

File: shell/platform/android/io/flutter/embedding/engine/android/FlutterFragment.java [hunk 1]
```diff
@@ -207,4 +209,102 @@ protected FlutterEngine onCreateFlutterEngine(@NonNull Context context) {
...
+   * Callback invoked when memory is low.
+   *
+   * This implementation forwards a memory pressure warning to the running Flutter app.
+   *
+   * @param level level
+   */
+  public void onTrimMemory(int level) {
+    // Use a trim level delivered while the application is running so the
+    // framework has a chance to react to the notification.
+    if (level == TRIM_MEMORY_RUNNING_LOW) {
```
Comment (dnfield) [inline shell/platform/android/io/flutter/embedding/engine/android/FlutterFragment.java]: Maybe not an issue for this PR, but I suspect we should be doing something with the CRITICAL version of this as well.  Maybe @jason-simmons has context?

**Comment by matthew-carroll** [comment]
@dnfield part of the reason for the TODOs is that different files are being filled in different PRs in parallel, so there is some cross dependency.  Also, I have been asked to keep these PRs around 100-200 lines long and moving some logic requires changes that might far exceed that.  So I'm doing what I can to move in brief pieces with rapid handling of TODOs to remove them after adding them.

**Comment by dnfield** [comment]
Would it make more sense to merge this one after FlutterEngine is in a place for it?

**Comment by matthew-carroll** [comment]
The bottleneck is the reviews, so we should merge anything that has been appropriately reviewed so that I can move on to handling the TODOs.

**Comment by dnfield** [comment]
Is there a different PR we could look to merge first so that more of this could be implemented?  This looks like it wouldn't really be useable in its current state.  Could we work through the PR to make FlutterEngine do what it needs to do to support this first?

**Comment by matthew-carroll** [comment]
@dnfield No, there is no other PR.  You're currently added to all open PRs and none of them fill out FlutterFragment.  I understand this is an unusual process, but it's simply the result of trying to migrate a staging branch as quickly as we can.

The goal of this PR is to add all the call forwarding that we know we'll need from FlutterActivity to FlutterFragment, regardless of what FlutterFragment does with it.  Do you have any review concerns about the javadocs on those methods, or the need for any of the FlutterActivity call forwards to exist?  If you believe the comments are appropriate, and the calls are necessary, then it would seem completely safe to move forward, right?

**Comment by matthew-carroll** [comment]
I'm going to force push after rebasing against master to pull in recent merged work.

**Comment by matthew-carroll** [comment]
@dnfield The functional TODOs have been resolved.  `FlutterFragment` still has more behavior to come, but I think the open-ended stuff should now be filled.

**Pull request merged**